### PR TITLE
[High][Admin|Category|Word|Choice] Text length validation and Specific Validation Messages

### PR DESCRIPTION
**Pull request by rdelrosariojr**
### [Asana Link]
**Text length validation**: https://app.asana.com/0/1201713837896454/1201933457103633/f
**Specific Validation Messages**: https://app.asana.com/0/1201713837896454/1201856632776564/f
### [Commands]
```
Back-end:
- `php artisan serve`

Front-end:
- `npm start`
```
### [Pre-conditions]
Make sure user rob@delrosario.com exist, otherwise, run `php artisan db:seed --class=UserTableSeeder`, then login with the following credentials:
email: rob@delrosario.com
password: password
### [Expected Output]
Correct error messages should be displayed
### [Notes]
Only Category Title, word, and choices has length validation. 
### [Screenshots]
image
image
image
image

### CODE REVIEW

File: front-end/e-learning/src/components/categories/Update.js [hunk 1]
```diff
@@ -42,11 +42,16 @@ const Update = ({
     if (title === "") {
       setErrorTitle("Field is required!");
```
Comment (AubreyV) [inline front-end/e-learning/src/components/categories/Update.js]: set to 'Title is required' here or no need na?

File: front-end/e-learning/src/components/categories/Create.js [hunk 2]
```diff
@@ -31,13 +31,18 @@ const Create = ({ createCategory, categories, token, user }) => {
   const validate = () => {
     var no_error = true;
     if (title === "") {
-      setErrorTitle("Field is required!");
+      setErrorTitle("Title is required!");
+      no_error = false;
+    } else if (title.length > 255) {
+      setErrorTitle("Title is limited to 255 characters!");
```
Comment (AubreyV) [inline front-end/e-learning/src/components/categories/Create.js:37]: **Not a comment, just note. It's up to you if you want to update now, but we can update this in the future.**
Next time, we can create a constants file, where we can store all constants, like these, error messages, default string length etc. Pros are, it's easier to update and easier to locate if we want to update.

example
```
// constants.js
errorMessages: {
  titleRequired: 'Title is required',
  titleMaxLength: 'Title is limited to 255 characters!',
  ...
}

export { errorMessages, ...<other constants if there are> }
```

then to use, you can just simply import the errorMessges constants file and make use of them to set error messages to the screen.

File: front-end/e-learning/src/components/categories/Update.js [hunk 3]
```diff
@@ -42,11 +42,16 @@ const Update = ({
...
       no_error = false;
+    } else if (title.length > 255) {
+      setErrorTitle("Title is limited to 255 characters!");
+      no_error = false;
     } else setErrorTitle("");
+
     if (description === "") {
       setErrorDescription("Field is required!");
```
Comment (AubreyV) [inline front-end/e-learning/src/components/categories/Update.js]: here should it be description? not sure please confirm.

File: front-end/e-learning/src/components/categories/Update.js (hunk 1 above)
Comment (rdelrosariojr) [reply front-end/e-learning/src/components/categories/Update.js]: @AubreyV I accidentally bypassed that one ma'am. I'll update it right away. Thank you.

File: front-end/e-learning/src/components/categories/Create.js (hunk 2 above)
Comment (rdelrosariojr) [reply front-end/e-learning/src/components/categories/Create.js:37]: @AubreyV yes ma'am, noted on this. Thank you for the insights, I'll bare this in mind.

**Review by AubreyV — APPROVED** [review]

**Pull request merged**